src/modules/powerrename/lib/Settings.cpp

```diff
@@ -470,12 +359,205 @@ void CRenameMRU::_FreeMRUList()
     }
 }
 
+CSettings::CSettings()
+{
+    std::wstring result = PTSettingsHelper::get_module_save_folder_location(L"PowerRename");
+    jsonFilePath = result + L"\\" + std::wstring(c_powerRenameDataFilePath);
+}
+
+bool CSettings::GetEnabled() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.enabled;
+}
+
+void CSettings::SetEnabled(bool enabled)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.enabled = enabled;
+}
+
+bool CSettings::GetShowIconOnMenu() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.showIconOnMenu;
+}
+
+void CSettings::SetShowIconOnMenu(bool show)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.showIconOnMenu = show;
+}
+
+bool CSettings::GetExtendedContextMenuOnly() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.extendedContextMenuOnly;
+}
+
+void CSettings::SetExtendedContextMenuOnly(bool extendedOnly)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.extendedContextMenuOnly = extendedOnly;
+}
+
+bool CSettings::GetPersistState() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.persistState;
+}
+
+void CSettings::SetPersistState(bool persistState)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.persistState = persistState;
+}
+
+bool CSettings::GetMRUEnabled() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.MRUEnabled;
+}
+
+void CSettings::SetMRUEnabled(bool MRUEnabled)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.MRUEnabled = MRUEnabled;
+}
+
+long CSettings::GetMaxMRUSize() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.maxMRUSize;
+}
+
+void CSettings::SetMaxMRUSize(long maxMRUSize)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.maxMRUSize = maxMRUSize;
+}
+
+long CSettings::GetFlags() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.flags;
+}
+
+void CSettings::SetFlags(long flags)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.flags = flags;
+}
+
+const std::wstring& CSettings::GetSearchText() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.searchText;
+}
+
+void CSettings::SetSearchText(const std::wstring& text)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.searchText = text;
+}
+
+const std::wstring& CSettings::GetReplaceText() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.replaceText;
+}
+
+void CSettings::SetReplaceText(const std::wstring& text)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.replaceText = text;
+}
+
+void CSettings::LoadPowerRenameData()
+{
+    std::scoped_lock lock{ dataLock };
+
+    if (!std::filesystem::exists(jsonFilePath))
+    {
+        MigrateSettingsFromRegistry();
+    }
+    else
+    {
+        ParseJsonSettings(GetPersistPowerRenameJSON());
+
+        SavePowerRenameData();

```

Comment:
move SavePowerRenameData() out of if-else, so we save after migration to?

Reply:
It actually belongs to if part, not here in else (when we do migration than we save to json).